### PR TITLE
Add configuration to segmentation IR

### PR DESCRIPTION
**Pull request by sovrasov**
### Summary
This PR adds ModelAPI-specific meta information into segmentation model IR.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by AlexKoff88 — DISMISSED** [review]

File: otx/algorithms/segmentation/utils/metadata.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    """Get ModelAPI config."""
+    omz_config = {}
+    omz_config[("model_info", "model_type")] = "segmentation"
+
+    omz_config[("model_info", "soft_threshold")] = hyperparams.postprocessing.soft_threshold
+    omz_config[("model_info", "blur_strength")] = hyperparams.postprocessing.blur_strength
+
+    all_labels = ""
+    for lbl in label_schema.get_labels(include_empty=False):
+        all_labels += lbl.name.replace(" ", "_") + " "
```
Comment (AlexKoff88) [inline otx/algorithms/segmentation/utils/metadata.py:22]: Hope this change won't cause any issues on the Geti side. Not sure who is the right person to comment here.

Comment (sovrasov) [reply otx/algorithms/segmentation/utils/metadata.py:22]: In the future, we have to assure that the process of conversion string labels to Geti's LabelEntity is reliable.

**Review by goodsong81 — APPROVED** [review]
Embedding itself here looks OK to me.
Would you somehow check whether the IR models from old OTX (Geti) without this segmentation metadata will be still working well without issues? I mean backward compatibility.

**Comment by sovrasov** [comment]
> Embedding itself here looks OK to me. Would you somehow check whether the IR models from old OTX (Geti) without this segmentation metadata will be still working well without issues? I mean backward compatibility.

This information is not used by OTX, so such issues can not arise (like we don't observe it in case of classification and detection).

**Comment by goodsong81** [comment]
> > Embedding itself here looks OK to me. Would you somehow check whether the IR models from old OTX (Geti) without this segmentation metadata will be still working well without issues? I mean backward compatibility.
> 
> This information is not used by OTX, so such issues can not arise (like we don't observe it in case of classification and detection).

Yes, I agree. When the new ModelAPIs will replace current OMZ ModelAPIs, we need to check such compatibility anyway. Thank you for clarification. :)

**Comment by sovrasov** [comment]
> > > Embedding itself here looks OK to me. Would you somehow check whether the IR models from old OTX (Geti) without this segmentation metadata will be still working well without issues? I mean backward compatibility.
> > 
> > 
> > This information is not used by OTX, so such issues can not arise (like we don't observe it in case of classification and detection).
> 
> Yes, I agree. When the new ModelAPIs will replace current OMZ ModelAPIs, we need to check such compatibility anyway. Thank you for clarification. :)

yeah, integration of ModelAPI and it's consequences for geti versions compatibility is a standalone discussion. It's complicated, since MAPI doesn't know about config.json, so we may need patch IRs in a JIT manner.

**Review by jaegukhyun — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**